prov/verbs/src/fi_verbs.c

```diff
@@ -2398,41 +2398,74 @@ static int fi_ibv_cq_reset(struct fid_cq *cq, const void *cond)
         return -ibv_req_notify_cq(_cq->cq, 0);
 }
 
+static inline int
+fi_ibv_poll_events(struct fi_ibv_cq *_cq, int timeout)
+{
+	int ret;
+	void *context;
+
+	ret = fi_poll_fd(_cq->channel->fd, timeout);
+	if (ret == 0)
+		return -FI_EAGAIN;
+	else if (ret < 0)
+		return ret;
+
+	ret = ibv_get_cq_event(_cq->channel, &_cq->cq, &context);
+	if (ret)
+		return ret;
+
+	ibv_ack_cq_events(_cq->cq, 1);
+
+	return 0;
+}
+
 static ssize_t
 fi_ibv_cq_sread(struct fid_cq *cq, void *buf, size_t count, const void *cond,
 		int timeout)
 {
 	ssize_t ret = 0, cur;
 	ssize_t  threshold;
-	int reset = 1;
 	struct fi_ibv_cq *_cq;
 
 	_cq = container_of(cq, struct fi_ibv_cq, cq_fid);
+
+	if (!_cq->channel)
+		return -FI_ENOSYS;
+
 	threshold = (_cq->wait_cond == FI_CQ_COND_THRESHOLD) ?
-		MIN((ssize_t) cond, count) : 1;
+		MIN((ssize_t) cond, count) : count;
```

Comment:
As the API is defined, yes.  The count is a maximum to the number of completions to report, not the minimum.  I can clarify this more in the man page.

Basically, fi_cq_sread with a timeout of 0 and no conditions should behave exactly like fi_cq_read